typed/typed.go

```diff
@@ -143,6 +153,12 @@ func (tv TypedValue) Compare(rhs TypedValue) (c *Comparison, err error) {
 }
 
 func merge(lhs, rhs TypedValue, rule, postRule mergeRule) (TypedValue, error) {
+	if !rhs.initialized() {
```

Comment:
> This changes behavior from a typed-but-empty object--the whole tree is still walked for those.

Does it? initialized only matches entirely non-initialized object that don't have typeref or schema. typed-but-empty objects are considered initialized and this won't change the behavior

Reply:
I'm going to add a test for that, but so far none of the tests have been impacted by this change (we should probably add more to be extra sure).

Reply:
I'm claiming if you send a real object with a typed-but-empty object, you'll find all the fields are walked and the compare function detects everything as added or deleted. But if you send with an uninitialized object, it would not list anything as added/deleted.

For that reason I don't think this should accept uninitialized objects, I think we should make it easy to make objects initialized with the right type.